utility.lua

```diff
@@ -70,38 +70,48 @@ local get_placer = function(player_name)
 end
 
 -- Used to remove items from the target block and put it into the hopper's inventory
-hopper.take_item_from = function(hopper_pos, target_pos, target_node, target_inventory_name)
-	if target_inventory_name == nil then
-		return
-	end
+hopper.take_item_from = function(hopper_pos, target_pos, target_node, target_inv_info)
 	local target_def = minetest.registered_nodes[target_node.name]
 	if not target_def then
 		return
 	end
 
+	local target_inv_name = target_inv_info["inventory_name"]
+	local target_get_inv  = target_inv_info["get_inventory"]
+
 	--hopper inventory
-	local hopper_meta = minetest.get_meta(hopper_pos);
+	local hopper_meta = minetest.get_meta(hopper_pos)
 	local hopper_inv = hopper_meta:get_inventory()
 	local placer = get_placer(hopper_meta:get_string("placer"))
 
 	--source inventory
-	local target_inv = minetest.get_meta(target_pos):get_inventory()
-	local target_inv_size = target_inv:get_size(target_inventory_name)
-	if target_inv:is_empty(target_inventory_name) == false then
+	local target_inv
+	if target_get_inv then
+		target_inv = target_get_inv(target_pos)
+		if not target_inv then
+			minetest.log("error","No inventory from api get_inventory function: " ..
+				minetest.get_meta(target_pos).name .. " on " .. vector.to_string(target_pos))
```

Comment:
Might not work. See other review comment.

Reply:
Fixed.